libMachO/MachO_ABI/LoadCommands/load_command_linker_option.c

```diff
@@ -0,0 +1,166 @@
+    .base.name                  = "LC_LINKER_OPTION",
+    .base.copy_description      = &_mk_load_command_linker_option_copy_description,
+    .command_id                 = LC_LINKER_OPTION,
+    .command_base_size          = sizeof(struct linker_option_command)
+};
+
+//|++++++++++++++++++++++++++++++++++++|//
+uint32_t mk_load_command_linker_option_id()
+{ return LC_LINKER_OPTION; }
+
+uint32_t
+mk_load_command_linker_option_get_nstrings(mk_load_command_ref load_command)
+{
+    _MK_LOAD_COMMAND_NOT_NULL(load_command, return UINT32_MAX);
+    _MK_LOAD_COMMAND_IS_A(load_command, _mk_load_command_linker_option_class, return UINT32_MAX);
+    
+    struct linker_option_command *linker_option_command = (struct linker_option_command*)load_command.load_command->mach_load_command;
+    return mk_macho_get_byte_order(load_command.load_command->image)->swap32( linker_option_command->count );
+}
+
+static const char*
+mk_load_command_linker_option_get_string(mk_load_command_ref load_command, uint32_t requested_index, size_t *string_length) {
+    uint32_t nstrings = mk_load_command_linker_option_get_nstrings(load_command);
+    if (nstrings == UINT32_MAX || requested_index >= nstrings) {
+        return NULL;
+    }
+
+    struct linker_option_command *linker_option_command = (struct linker_option_command*)load_command.load_command->mach_load_command;
+    char *command_end = ((char*)linker_option_command) + linker_option_command->cmdsize;
+    char *current_option = ((char*)linker_option_command) + sizeof(struct linker_option_command);
+    uint32_t current_index = 0;
+    
+    while (current_index <= requested_index && current_option < command_end) {
+        size_t max_len = (size_t)(command_end - current_option);
+        size_t current_string_len = strnlen(current_option, max_len);
+        if (current_string_len == max_len) {
+            // By the command spec, all strings must be NULL terminated
+            return NULL;
+        }
+        
+        if (current_index == requested_index) {
+            if (string_length) {
+                *string_length = current_string_len;
+            }
+            return current_option;
+        }
+        
+        ++current_index;
+        current_option += current_string_len + 1 /* NULL char */;
+    }
+    
+    // Last processed NULL-terminated string ends exactly at the command end
+    return NULL;
+}
+
+size_t
+mk_load_command_linker_option_copy_string(mk_load_command_ref load_command, uint32_t requested_index, char *output, size_t output_len)
+{
+    _MK_LOAD_COMMAND_NOT_NULL(load_command, return 0);
+    _MK_LOAD_COMMAND_IS_A(load_command, _mk_load_command_linker_option_class, return 0);
+    
+    size_t current_string_len = 0;
+    const char *current_option_start = mk_load_command_linker_option_get_string(load_command, requested_index, &current_string_len);
+    
+    if ((output == NULL) || (current_option_start != NULL && (current_string_len + 1 /* NULL char */) <= output_len)) {
```

Comment:
This still does not match the semantics of the other APIs that copy out strings.  See then end of `_mk_mach_lc_str_copy` (called by `mk_load_command_rpath_copy_path`) for an example.

If the size of `output_len` is < `current_string_len`, then `output_len - 1` bytes should be copied, and the last byte set to the `NULL` terminator`.

Reply:
@DeVaukz: No problem. The exact desired semantics were not clear to me beforehand, I've now adjusted them to match as requested (note: I've just updated the last commit in the PR).